[BUG] Range Partitioning can crash when processing is in the order-by

**Describe the bug**
If we are going to order by a column that does not exist, but is generated on the fly Rnge partitioning crashes.

**Steps/Code to reproduce bug**
I am adding some tests as a part of the out of core sort work, but the diff is here...

```diff
diff --git a/integration_tests/src/main/python/sort_test.py b/integration_tests/src/main/python/sort_test.py
index 7c1ebd2a..6cf0003c 100644
--- a/integration_tests/src/main/python/sort_test.py
+++ b/integration_tests/src/main/python/sort_test.py
@@ -67,3 +67,15 @@ orderable_gens_sort_without_neg_decimal = [n for n in orderable_gens_sort if not
 def test_multi_orderby_with_limit(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : binary_op_df(spark, data_gen).orderBy(f.col('a'), f.col('b').desc()).limit(100))
+
+# We are not trying all possabilities, just doing a few where the numbers should work out.
+@pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
+def test_orderby_with_processing(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')))
+
+# We are not trying all possabilities, just doing a few where the numbers should work out.
+@pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
+def test_orderby_with_processing_and_limit(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')).limit(100))
```

The first test fails.  The second test passes
```
E                   Caused by: java.lang.UnsupportedOperationException: Cannot evaluate expression: (100.0 - input[0, float, true])
E                   	at org.apache.spark.sql.catalyst.expressions.Unevaluable.eval(Expression.scala:304)
E                   	at org.apache.spark.sql.catalyst.expressions.Unevaluable.eval$(Expression.scala:303)
E                   	at com.nvidia.spark.rapids.CudfBinaryOperator.eval(GpuExpressions.scala:261)
E                   	at org.apache.spark.sql.catalyst.expressions.InterpretedUnsafeProjection.apply(InterpretedUnsafeProjection.scala:90)
E                   	at org.apache.spark.sql.catalyst.expressions.InterpretedUnsafeProjection.apply(InterpretedUnsafeProjection.scala:34)
E                   	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
E                   	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
E                   	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
E                   	at com.nvidia.spark.rapids.SamplingUtils$.reservoirSampleAndCount(SamplingUtils.scala:47)
E                   	at com.nvidia.spark.rapids.GpuRangePartitioner.$anonfun$sketch$1(GpuRangePartitioner.scala:49)
E                   	at com.nvidia.spark.rapids.GpuRangePartitioner.$anonfun$sketch$1$adapted(GpuRangePartitioner.scala:46)
E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2(RDD.scala:915)
E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2$adapted(RDD.scala:915)
E                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
E                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
E                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
E                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
E                   	at org.apache.spark.scheduler.Task.run(Task.scala:131)
E                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
E                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
E                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
E                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
E                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
E                   	... 1 more
```

**Expected behavior**
We should be able to sort just like the CPU does.

**Additional context**
For a perminent fix we might want to look at combining this work with #1727.  In the short term we can probably disable this use case (as it has not shown up outside of testing) and fall back to the CPU.

--- a/integration_tests/src/main/python/sort_test.py
+++ b/integration_tests/src/main/python/sort_test.py
@@ -67,3 +67,15 @@ orderable_gens_sort_without_neg_decimal = [n for n in orderable_gens_sort if not
 def test_multi_orderby_with_limit(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : binary_op_df(spark, data_gen).orderBy(f.col('a'), f.col('b').desc()).limit(100))
+
+# We are not trying all possabilities, just doing a few where the numbers should work out.
+@pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
+def test_orderby_with_processing(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')))
+
+# We are not trying all possabilities, just doing a few where the numbers should work out.
+@pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
+def test_orderby_with_processing_and_limit(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')).limit(100))

--- a/integration_tests/src/main/python/sort_test.py
+++ b/integration_tests/src/main/python/sort_test.py
@@ -68,13 +68,13 @@ def test_multi_orderby_with_limit(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : binary_op_df(spark, data_gen).orderBy(f.col('a'), f.col('b').desc()).limit(100))
 
-# We are not trying all possabilities, just doing a few where the numbers should work out.
+# We are not trying all possibilities, just doing a few with numbers so the query works.
 @pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
 def test_orderby_with_processing(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')))
 
-# We are not trying all possabilities, just doing a few where the numbers should work out.
+# We are not trying all possibilities, just doing a few with numbers so the query works.
 @pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
 def test_orderby_with_processing_and_limit(data_gen):
     assert_gpu_and_cpu_are_equal_collect(

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala
@@ -2386,6 +2386,18 @@ object GpuOverrides {
         override val childExprs: Seq[BaseExprMeta[_]] =
           rp.ordering.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
 
+        override def tagPartForGpu(): Unit = {
+          def isSortOrderSimpleEnough(so: SortOrder): Boolean = so.child match {
+            case _: AttributeReference => true
+            case _ => false
+          }
+          // Once https://github.com/NVIDIA/spark-rapids/issues/1730 is fixed this check should be
+          // removed
+          if (!rp.ordering.forall(isSortOrderSimpleEnough)) {
+            willNotWorkOnGpu("computation is not supported for sort order in range partitioning")
+          }
+        }
+
         override def convertToGpu(): GpuPartitioning = {
           if (rp.numPartitions > 1) {
             val gpuOrdering = childExprs.map(_.convertToGpu()).asInstanceOf[Seq[SortOrder]]

